jupyter_server_fileid/manager.py

```diff
@@ -395,12 +422,35 @@ def __init__(self, *args, **kwargs):
         self.con.commit()
 
     def _normalize_path(self, path):
-        path = super()._normalize_path(path)
+        """Accepts an API path and returns a filesystem path, i.e. one prefixed
+        by root_dir and uses os.path.sep."""
+        # use commonprefix instead of commonpath, since root_dir may not be a
+        # absolute POSIX path.
+        if os.path.commonprefix([self.root_dir, path]) != self.root_dir:
+            path = os.path.join(self.root_dir, path)
+
```

Comment:
You can remove the comment and use `os.path.commonpath()` here since LocalFileIdManager only deals with local filesystem paths.

Reply:
Turns out changing this to `os.path.commonpath()` causes failures with `ValueError: Can't mix absolute and relative paths` when the `root_dir` is: '/private/var/folders/t0/lmgg5t1j35dcftfz88jf1mqw0000gn/T/pytest-of-kbates/pytest-114/test_delete_recursive_local_0/root_dir' and the `path` is 'test_path'.

Since this change isn't really germane to this PR (this code was already this way, just located in the superclass that LFIDMgr calls), I'll leave this alone due to all the failures it introduces - even on MacOS tests.

Reply:
Bizarre! Not sure why that is; we were using `os.path.commonpath()` in the previous implementation.